src/Symfony/Component/Mailer/EventListener/EnvelopeListener.php

```diff
@@ -30,17 +31,25 @@ class EnvelopeListener implements EventSubscriberInterface
      */
     private ?array $recipients = null;
 
+    /**
+     * @var string[]
+     */
+    private array $allowedRecipients;
+
     /**
      * @param array<Address|string> $recipients
+     * @param string[]              $allowedRecipients An array of regex to match the allowed recipients
      */
-    public function __construct(Address|string|null $sender = null, ?array $recipients = null)
+    public function __construct(Address|string|null $sender = null, ?array $recipients = null, ?array $allowedRecipients = null)
```

Comment:
You pass `[]` anyway:
```suggestion
    public function __construct(Address|string|null $sender = null, ?array $recipients = null, array $allowedRecipients = [])
```